### PR TITLE
Ui profile follow button

### PR DESCRIPTION
**Pull request by bujars**
This is the pull request for the Follow/Unfollow button (UI) for a users profile. Note focus/comments should be on your_id (ie how are we going to distinguish the user from whose profile we are using, and how the state/button renders.

We will also need backend for this (down the line) 

Tests focus on prototypes, and the state value before and after change, and which Button Follow/Unfollow is rendered depending on the state.

### CODE REVIEW

File: src/screens/app/Components/CommentsTouchBox/CommentsTouchBox.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import React from 'react';
-import {TouchableWithoutFeedback} from 'react-native-gesture-handler';
 import {Caption} from 'react-native-paper';
-import {StyleSheet} from 'react-native';
+import {StyleSheet, TouchableOpacity} from 'react-native';
```
Comment (bujars) [inline src/screens/app/Components/CommentsTouchBox/CommentsTouchBox.js:4]: Previous pull request. Don't also comment here.

File: __tests__/ProfileBanner-test.js [hunk 2]
```diff
@@ -5,16 +5,47 @@ import ProfileBanner from '@screens/app/Components/ProfileBanner/ProfileBanner';
...
+        posts_count={0}
+        follower_count={0}
+        following_count={0}
+        profile_image={''}
+        isFollowing={false}
+        your_id={'b'}
+      />,
+    )
+    .toJSON();
+  expect(treeNotMatchingUsernameFalseFollowing).toMatchSnapshot();
```
Comment (EramM805) [inline __tests__/ProfileBanner-test.js:50]: Do follower_count, posts_count, and following_count value > 0 do another behavior, if yes, we should write a snapshot test for it

File: __tests__/RequestFollowButton-test.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+function testPropTypesWithErrors(isFollowing, errorMsg, testDescription) {
+  it('Throws failed propType Error for ' + testDescription, () => {
+    let result = checkPropTypes(
+      RequestFollowButton.propTypes,
+      {isFollowing: isFollowing},
+      'prop',
+      RequestFollowButton.isFollowing,
+    );
+    console.log(errorMsg);
```
Comment (EramM805) [inline __tests__/RequestFollowButton-test.js:89]: Please remove console.logs

File: src/screens/app/Components/RequestFollowButton/RequestFollowButton.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  removeFollower = () => {
+    /** Set the state to no longer following, and do backend to remove! */
+    this.setState({isFollowing: false});
+    console.log('Follower Removed!');
+    /** Do componenets re-render when state changes? I believe so.... */
+  };
+
+  addFollower = () => {
+    this.setState({isFollowing: true});
+    console.log('Follower Added!');
```
Comment (EramM805) [inline src/screens/app/Components/RequestFollowButton/RequestFollowButton.js:29]: Again should add backend code to this

File: src/screens/app/Components/ProfileImageList/ProfileImageList.js [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+      {
+        userName: 'Model1234',
+        userImage: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80',
+        comment: 'Slay girllll!!!'
+      }
+    ], 
+    
+    img: 'https://www.familyandmedia.eu/wp-content/uploads/2018/10/529382-4386816-selfie-psicologia-725x545.jpg', 
+    date_posted: '10-10-2019'
+  },
```
Comment (coollama1) [inline src/screens/app/Components/ProfileImageList/ProfileImageList.js]: Consider automating post creation process with a function

File: __tests__/ProfileImageList-test.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    instance.setState({
+      profile_view: 'grid',
+    }); /** WE just want to check the grid view first. (Force value) */
+    /** get the current state -- for what reason? */
+    //const viewState = instance.state.profile_view;
+    //expect(viewState).toEqual('grid');
+
+    const flatlistWrapper = wrapper.find(FlatList);
+    //const threeColWrapper = flatlistWrapper.find({numColumns: 3});
+    //console.log(threeColWrapper.first().props().numColumns);
```
Comment (coollama1) [inline __tests__/ProfileImageList-test.js:95]: If comments aren't necessary, consider deleting them

File: __tests__/ProfileImageList-test.js [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+    expect(flatlistWrapper.find({numColumns: 3})).toHaveLength(0);
+  });
+}
+
+testWhichFlatlistShows();
+
+// function testOnViewHeadlineIconButtonClicked(initalProfileViewStatus){
```
Comment (coollama1) [inline __tests__/ProfileImageList-test.js:138]: Consider deleting unnecessary comments. Previous code can be recovered from past commits

File: __tests__/ProfileBanner-test.js [hunk 8]
```diff
@@ -5,16 +5,47 @@ import ProfileBanner from '@screens/app/Components/ProfileBanner/ProfileBanner';
...
+        username={'a'}
+        posts_count={0}
+        follower_count={0}
+        following_count={0}
+        profile_image={''}
+        isFollowing={true}
+        your_id={'b'}
+      />,
+    )
+    .toJSON();
```
Comment (coollama1) [inline __tests__/ProfileBanner-test.js:34]: Consider automating ProfileBanner creation with the use of functions for object/component creation

File: __tests__/ProfileBanner-test.js [hunk 9]
```diff
@@ -5,16 +5,47 @@ import ProfileBanner from '@screens/app/Components/ProfileBanner/ProfileBanner';
...
 test('Profile Banner Snapshot Test', () => {
-  const tree = renderer
+  const treeMatchingUsername = renderer
     .create(
       <ProfileBanner
         username={''}
         posts_count={0}
         follower_count={0}
         following_count={0}
         profile_image={''}
```
Comment (kartikeya01) [inline __tests__/ProfileBanner-test.js:15]: are we gonna have a default profile image?

File: __tests__/ProfileBanner-test.js (hunk 2 above)
Comment (bujars) [reply __tests__/ProfileBanner-test.js:50]: To Do in the future Sprint. Not of strict importance

File: src/screens/app/Components/RequestFollowButton/RequestFollowButton.js (hunk 4 above)
Comment (bujars) [reply src/screens/app/Components/RequestFollowButton/RequestFollowButton.js:29]: Left for Backend to do

**Pull request merged**